pandas/core/indexes/base.py

```diff
@@ -3317,7 +3317,7 @@ def _join_monotonic(self, other, how='left', return_indexers=False):
 
     def _wrap_joined_index(self, joined, other):
         name = self.name if self.name == other.name else None
-        return Index(joined, name=name)
+        return self.__class__(joined, name=name)
```

Comment:
ok.  thought `_wrap_union_result` uses the `self.__class__` approach so I think this should be ok.